src/AutoRest.CSharp/Mgmt/Models/OperationSet.cs

```diff
@@ -40,6 +40,18 @@ public OperationSet(string requestPath, InputClient? inputClient)
             Operations = new HashSet<InputOperation>();
         }
 
+        public bool IsNonResourceOperations(InputNamespace? inputNamespace)
+        {
+            // We use decorator @service to tell if it is a TypeSpec input
+            bool isTsp = inputNamespace?.Clients.FirstOrDefault(c => c.Decorators.Any(d => d.Name == "TypeSpec.@service")) != null;
+            // If swagger input, we don't use this operation to check if it is a non resource operation
+            if (!isTsp)
+                return false;
```

Comment:
we could add a `IsTypeSpec` flag on `MgmtContext`, which is static so that we could get the result and cache it there.

Reply:
well we might have a concurrent issue when running test cases if we do this